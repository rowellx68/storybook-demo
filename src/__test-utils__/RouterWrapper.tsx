import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';

const RouterWrapper: React.FunctionComponent = ({ children }) => {
  return (
    <MemoryRouter keyLength={0}>
      <Route exact path='/' />
      {children}
    </MemoryRouter>
  );
};

export default RouterWrapper;
