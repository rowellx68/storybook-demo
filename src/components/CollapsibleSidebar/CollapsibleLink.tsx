/** @jsx jsx */
import React from 'react';
import { NavLink } from 'react-router-dom';
import { jsx } from '@emotion/core';

import { button } from './collapsible-shared-styles';

interface LinkProps {
  expanded: boolean;
  icons: string[];
  to: string;
  exact?: boolean;
  label?: string;
}

const CollapsibleLink: React.FunctionComponent<LinkProps> = ({
  expanded,
  icons,
  to,
  exact = true,
  label
}) => {
  return (
    <NavLink exact={exact} title={label} to={to} css={button}>
      {expanded ? (
        <React.Fragment>
          <i className={icons[0]} aria-hidden='true'></i>
          <span>{label}</span>
        </React.Fragment>
      ) : (
        <i className={icons[1] || icons[0]} aria-hidden='true'></i>
      )}
    </NavLink>
  );
};

export default CollapsibleLink;
