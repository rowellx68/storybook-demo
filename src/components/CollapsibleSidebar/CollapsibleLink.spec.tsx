import React from 'react';
import { mount } from 'enzyme';

import RouterWrapper from '__test-utils__/RouterWrapper';
import CollapsibleLink from 'components/CollapsibleSidebar/CollapsibleLink';

const options = {
  wrappingComponent: RouterWrapper,
};

const icons = ['las la-fw la-ship'];

describe('CollapsibleLink', () => {
  it('renders the link', () => {
    const component = mount(
      <CollapsibleLink
        expanded={false}
        icons={icons}
        label='Projects'
        to='/projects'
      />,
      options,
    );

    expect(component.exists('a')).toEqual(true);
  });

  it('renders the correct mark-up - collapsed', () => {
    const component = mount(
      <CollapsibleLink
        expanded={false}
        icons={icons}
        label='Projects'
        to='/projects'
      />,
      options,
    );

    expect(component).toMatchSnapshot();
  });

  it('renders the label', () => {
    const component = mount(
      <CollapsibleLink expanded={true} icons={icons} label='Projects' to='/' />,
      options,
    );

    expect(component).toContainReact(<span>Projects</span>);
  });

  it('renders the correct mark-up - expanded', () => {
    const component = mount(
      <CollapsibleLink
        expanded={true}
        icons={icons}
        label='Projects'
        to='/projects'
      />,
      options,
    );

    expect(component).toMatchSnapshot();
  });
});
