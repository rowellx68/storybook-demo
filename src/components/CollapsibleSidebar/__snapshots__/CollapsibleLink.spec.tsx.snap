// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleLink renders the correct mark-up - collapsed 1`] = `
.emotion-0 {
  padding: 0.5rem;
  border-radius: 0.25rem;
  border: none;
  background: #f5f6fa;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #4a5568;
}

.emotion-0:hover {
  background: rgba(0,0,0,0.1);
}

@media screen and (min-width:640px) {
  .emotion-0 {
    margin-bottom: 0.5rem;
  }

  .emotion-0:last-of-type {
    margin-bottom: 0;
  }
}

.emotion-0.active {
  color: var(--primary);
  background: #ffffff;
}

.emotion-0 i {
  font-size: 1.5rem;
  opacity: 1;
}

.emotion-0 span {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1rem;
  padding: 0 0.75rem;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto', 'Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans', 'Helvetica Neue',sans-serif;
}

@media screen and (min-width:640px) {
  .emotion-0 span {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<CollapsibleLink
  expanded={false}
  icons={
    Array [
      "las la-fw la-ship",
    ]
  }
  label="Projects"
  to="/projects"
>
  <NavLink
    className="emotion-0"
    exact={true}
    title="Projects"
    to="/projects"
  >
    <Link
      aria-current={null}
      className="emotion-0"
      title="Projects"
      to={
        Object {
          "hash": "",
          "pathname": "/projects",
          "search": "",
          "state": null,
        }
      }
    >
      <LinkAnchor
        aria-current={null}
        className="emotion-0"
        href="/projects"
        navigate={[Function]}
        title="Projects"
      >
        <a
          aria-current={null}
          className="emotion-0"
          href="/projects"
          onClick={[Function]}
          title="Projects"
        >
          <i
            aria-hidden="true"
            className="las la-fw la-ship"
          />
        </a>
      </LinkAnchor>
    </Link>
  </NavLink>
</CollapsibleLink>
`;

exports[`CollapsibleLink renders the correct mark-up - expanded 1`] = `
.emotion-0 {
  padding: 0.5rem;
  border-radius: 0.25rem;
  border: none;
  background: #f5f6fa;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #4a5568;
}

.emotion-0:hover {
  background: rgba(0,0,0,0.1);
}

@media screen and (min-width:640px) {
  .emotion-0 {
    margin-bottom: 0.5rem;
  }

  .emotion-0:last-of-type {
    margin-bottom: 0;
  }
}

.emotion-0.active {
  color: var(--primary);
  background: #ffffff;
}

.emotion-0 i {
  font-size: 1.5rem;
  opacity: 1;
}

.emotion-0 span {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1rem;
  padding: 0 0.75rem;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto', 'Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans', 'Helvetica Neue',sans-serif;
}

@media screen and (min-width:640px) {
  .emotion-0 span {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<CollapsibleLink
  expanded={true}
  icons={
    Array [
      "las la-fw la-ship",
    ]
  }
  label="Projects"
  to="/projects"
>
  <NavLink
    className="emotion-0"
    exact={true}
    title="Projects"
    to="/projects"
  >
    <Link
      aria-current={null}
      className="emotion-0"
      title="Projects"
      to={
        Object {
          "hash": "",
          "pathname": "/projects",
          "search": "",
          "state": null,
        }
      }
    >
      <LinkAnchor
        aria-current={null}
        className="emotion-0"
        href="/projects"
        navigate={[Function]}
        title="Projects"
      >
        <a
          aria-current={null}
          className="emotion-0"
          href="/projects"
          onClick={[Function]}
          title="Projects"
        >
          <i
            aria-hidden="true"
            className="las la-fw la-ship"
          />
          <span>
            Projects
          </span>
        </a>
      </LinkAnchor>
    </Link>
  </NavLink>
</CollapsibleLink>
`;
